pkg/noderesourcetopology/filter_test.go

```diff
@@ -0,0 +1,461 @@
+						Name:        nicResourceName,
+						Capacity:    intstr.Parse("30"),
+						Allocatable: intstr.Parse("5"),
+					},
+				},
+			}, topologyv1alpha1.Zone{
+				Name: "node-1",
+				Type: "Node",
+				Resources: topologyv1alpha1.ResourceInfoList{
+					topologyv1alpha1.ResourceInfo{
+						Name:        "cpu",
+						Capacity:    intstr.Parse("30"),
+						Allocatable: intstr.Parse("4"),
+					}, topologyv1alpha1.ResourceInfo{
+						Name:        "memory",
+						Capacity:    intstr.Parse("8Gi"),
+						Allocatable: intstr.Parse("4Gi"),
+					}, topologyv1alpha1.ResourceInfo{
+						Name:        nicResourceName,
+						Capacity:    intstr.Parse("30"),
+						Allocatable: intstr.Parse("2"),
+					},
+				},
+			},
+		},
+	}
+
+	nodes["default/node3"] = topologyv1alpha1.NodeResourceTopology{
+		ObjectMeta:       metav1.ObjectMeta{Name: "node3"},
+		TopologyPolicies: []string{string(topologyv1alpha1.SingleNUMANodePodLevel)},
+		Zones: topologyv1alpha1.ZoneList{
+			topologyv1alpha1.Zone{
+				Name: "node-0",
+				Type: "Node",
+				Resources: topologyv1alpha1.ResourceInfoList{
+					topologyv1alpha1.ResourceInfo{
+						Name:        "cpu",
+						Capacity:    intstr.Parse("20"),
+						Allocatable: intstr.Parse("2"),
+					}, topologyv1alpha1.ResourceInfo{
+						Name:        "memory",
+						Capacity:    intstr.Parse("8Gi"),
+						Allocatable: intstr.Parse("4Gi"),
+					}, topologyv1alpha1.ResourceInfo{
+						Name:        nicResourceName,
+						Capacity:    intstr.Parse("30"),
+						Allocatable: intstr.Parse("5"),
+					},
+				},
+			}, topologyv1alpha1.Zone{
+				Name: "node-1",
+				Type: "Node",
+				Resources: topologyv1alpha1.ResourceInfoList{
+					topologyv1alpha1.ResourceInfo{
+						Name:        "cpu",
+						Capacity:    intstr.Parse("30"),
+						Allocatable: intstr.Parse("4"),
+					}, topologyv1alpha1.ResourceInfo{
+						Name:        "memory",
+						Capacity:    intstr.Parse("8Gi"),
+						Allocatable: intstr.Parse("4Gi"),
+					}, topologyv1alpha1.ResourceInfo{
+						Name:        nicResourceName,
+						Capacity:    intstr.Parse("30"),
+						Allocatable: intstr.Parse("2"),
+					},
+				},
+			},
+		},
+	}
+	node1Resources := makeResourceListFromZones(nodes["default/node1"].Zones)
+	node1 := v1.Node{
+		ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+		Status: v1.NodeStatus{
+			Capacity:    node1Resources,
+			Allocatable: node1Resources,
+		},
+	}
+
+	node2Resources := makeResourceListFromZones(nodes["default/node2"].Zones)
+	node2 := v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "node2"}, Status: v1.NodeStatus{
+		Capacity:    node2Resources,
+		Allocatable: node2Resources,
+	},
+	}
+
+	node3Resources := makeResourceListFromZones(nodes["default/node3"].Zones)
+
+	node3 := v1.Node{ObjectMeta: metav1.ObjectMeta{Name: "node3"}, Status: v1.NodeStatus{
+		Capacity:    node3Resources,
+		Allocatable: node3Resources,
+	},
+	}
+
+	// Test different QoS Guaranteed/Burstable/BestEffort
+	topologyTests := topologyMatchTests{
```

Comment:
It's more conventional to make a single struct `topologyMatchTest` and build the slice here `topologyTests := []topologyMatchTest`